runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -671,10 +671,10 @@ protected void unmapSecondaryIndexes(K key, V value) {
     @SuppressWarnings("unchecked")
     protected void mapSecondaryIndexes(K key, V value) {
         try {
-            Map.Entry<K, V> entry = new AbstractMap.SimpleImmutableEntry<>(key, value);
             indexFunctions.parallelStream()
                     .forEach(f -> f.getIndexFunction().generateIndex(key, value)
-                            .forEach(i -> indexMap.get(f).put((I) i, entry)));
+                            .forEach(i -> indexMap.get(f).computeIfAbsent((I) i,
+                                    (I) -> new HashMap<K, V>()).put(key, value)));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "I" parameter [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1611)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)